Adding recursive terminate/purge in workflow API (#3969)
dapr/docs: daprdocs/content/en/reference/api/workflow_api.md

Terminate a running workflow instance with the given name and instance ID.

```
POST http://localhost:3500/v1.0-beta1/workflows/<workflowComponentName>/<instanceId>/terminate[?non_recursive=false]
```

{{% alert title="Note" color="primary" %}}

>  Terminating a workflow will terminate all of the child workflows created by the workflow instance. This can be disabled by setting the query parameter `non_recursive` to `true`. 

Reviewer: ```suggestion
 Terminating a workflow terminates all of the child workflows created by the workflow instance. You can disable this by setting the query parameter `non_recursive` to `true`. 
```

>  Terminating a workflow will have no effect on any in-flight activity executions that were started by the terminated instance. 

Reviewer: ```suggestion
Terminating a workflow has no effect on any in-flight activity executions that were started by the terminated instance. 
```

{{% /alert %}}

### URL parameters

Parameter | Description
--------- | -----------
`workflowComponentName` | Use `dapr` for Dapr Workflows
`instanceId` | Unique value created for each run of a specific workflow

> `non_recursive` | (Optional) Boolean to determine if Dapr should not recursively terminate child workflows created by the workflow instance, default value is false.

Reviewer: ```suggestion
`non_recursive` | (Optional) Boolean to determine if Dapr should not recursively terminate child workflows created by the workflow instance. Default value is `false`.
```

Purge the workflow state from your state store with the workflow's instance ID.

```
POST http://localhost:3500/v1.0-beta1/workflows/<workflowComponentName>/<instanceId>/purge[?non_recursive=false]
```

{{% alert title="Note" color="primary" %}}

>  Purging a workflow will purge all of the child workflows created by the workflow instance. This can be disabled by setting the query parameter `non_recursive` to `true`.

Reviewer: ```suggestion
 Purging a workflow purges all of the child workflows created by the workflow instance. You can disable this by setting the query parameter `non_recursive` to `true`.
```

{{% /alert %}}

### URL parameters

Parameter | Description
--------- | -----------
`workflowComponentName` | Use `dapr` for Dapr Workflows
`instanceId` | Unique value created for each run of a specific workflow

> `non_recursive` | (Optional) Boolean to determine if Dapr should not recursively purge child workflows created by the workflow instance, default value is false.

Reviewer: ```suggestion
`non_recursive` | (Optional) Boolean to determine if Dapr should not recursively purge child workflows created by the workflow instance. Default value is `false`.
```
